Sources/TuistCore/Cache/CacheOutputType.swift

```diff
@@ -1,24 +1,33 @@
 import Foundation
 
 /// An enum that represents the type of output that the caching feature can work with.
-public enum CacheOutputType: CustomStringConvertible {
+public enum CacheOutputType: CustomStringConvertible, Equatable {
     /// Resource bundle built for the simulator
     case bundle
 
     /// Frameworks built for the simulator.
     case framework
 
-    /// XCFrameworks built for the simulator and device.
-    case xcframework
+    /// XCFrameworks built for the simulator and/or device.
+    case xcframework(CacheXCFrameworkDestination)
 
     public var description: String {
         switch self {
         case .bundle:
             return "bundle"
         case .framework:
             return "framework"
-        case .xcframework:
-            return "xcframework"
+        case let .xcframework(destination):
+            switch destination {
+            case [.device, .simulator]:
+                return "xcframework"
+            case .device:
+                return "device-xcframework"
+            case .simulator:
+                return "simulator-xcframework"
+            default:
+                fatalError("xcframework should contain at least one destination")
```

Comment:
This should never happen, right? It might be better to throw an error but if we're sure this condition will never happen, I don't necessarily mind the `fatalError` here

Reply:
yes it should never happen, as there is a validation in the command argument parser